api/v1/routes/contact_us.py

```diff
@@ -0,0 +1,48 @@
+import logging
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel, EmailStr, constr
+from fastapi.responses import JSONResponse
+import requests
+
+
+router = APIRouter()
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class ContactForm(BaseModel):
+    name: constr(strip_whitespace=True, min_length=1)
+    email: EmailStr
+    message: constr(strip_whitespace=True, min_length=1)
+
+def send_email_via_service(name: str, email: str, message: str) -> bool:
+    url = "http://localhost:8000/api/v1/send-mail"
+    headers = {
```

Comment:
The `url` is not needed, emailing functionality would be a service within the app

Reply:
I have implemented like you advised. I have presumed that the email service file is in the services directory and I have used the following code as my email_services to simulate that my code is working properly.

```
import os
import requests
import logging
from dotenv import load_dotenv

load_dotenv()

# Setup logging
logger = logging.getLogger(__name__)


class EmailService:
    def __init__(self):
        self.api_key = os.getenv('BREVO_API_KEY')
        self.receiver_email = os.getenv('RECEIVER_EMAIL')

    def send_email(self, name: str, email: str, message: str) -> bool:
        url = "https://api.brevo.com/v3/smtp/email"
        headers = {
                "accept": "application/json",
                "api-key": self.api_key,
                "content-type": "application/json"
        }
        data = {
                "sender": {"name": name, "email": email},
                "to": [{"email": self.receiver_email}],
                "subject": "New Contact Us Inquiry",
                "htmlContent": f"<p>Name: {name}</p><p>Email: {email}</p><p>Message: {message}</p>"
        }

        try:
            response = requests.post(url, json=data, headers=headers)
            response.raise_for_status()
            logger.info(f"Email sent successfully to {self.receiver_email}")
            return True
        except requests.exceptions.HTTPError as err:
            logger.error(f"Failed to send email: {err}")
            return False
```